### ISSUE TITLE
Allow querying HGIG HDR metadata for outputs

### ISSUE THREAD

**Issue by magcius** [KhronosGroup/Vulkan-Docs#2167]
The [HGIG recommendations](http://hgig.org/doc/ForBetterHDRGaming.pdf#page=11) suggest exposing three fields to game developers to let them best do tonemapping in HDR: MaxFullFrameToneMapLuminance (MaxFFTML), MaxToneMapLuminance (MaxTML) and MinToneMapLuminance (MinTML).

DXGI already supports [querying these values from an IDXGIOutput6](https://learn.microsoft.com/en-us/windows/win32/api/dxgi1_6/ns-dxgi1_6-dxgi_output_desc1), however Vulkan doesn't seem to expose the same functionality yet. Thus, it's difficult to retrieve the display-specific parameters for correct HDR tonemapping.

Note that Vulkan does have support for setting [application-provided SMPTE ST.2086 HDR metadata](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkHdrMetadataEXT.html), but this approach is discouraged, because TVs and display environments have not provided consistent display tonemapping in response to this metadata. See the note / remarks on the [equivalent DXGI functionality](https://learn.microsoft.com/en-us/windows/win32/api/dxgi1_5/nf-dxgi1_5-idxgiswapchain4-sethdrmetadata)

**Comment by versalinyaa** [comment]
Thanks @magcius . The Vulkan System Integration TSG will begin discussing this tomorrow.

**Comment by alecazam** [comment]
The hdr metadata extension seems undefined as to it's application.  AMD only sets this in a FSE HDR environment where the app has taken over control of the display.   But Win10/11 have moved to borderless window now, so that it works more like macOS and the gui continues to work.

**Comment by versalinyaa** [comment]
@alecazam IIUC, are you saying that `VK_EXT_hdr_metadata` no longer works at all in Windows 10/11?

**Comment by alecazam** [comment]
I have no idea. I haven't seen the extension do anything in a windowed or borderless windowed config.   DXGI has already established those settings, and so I assume the Vulkan swapchain just sends off our HDR10 data directly to the DWM, or using the fast-flip model directly to DXGI and to the display.  HDR seems to be working correctly.

The docs on this extension say to look at the spec, and then the spec says that details on the extension are outside of the Vulkan spec.  And there is no HDR Vulkan sample trying to handle HDR on multi-mon on Windows.   Most indicate that the entire Vulkan swapchain should be replaced with a shared DXGI framebuffer.

I assume this extension only applies to Fullscreen Exclusive which has been phased out by Microsoft.  Windows Store apps can't change monitor resolution or refresh.  The system (Windows and gpu control panel) should be handling all this.  But again without real documentation or samples, I have no idea what this extension even does.  DXGI supplies all the data needed to configure this.

**Comment by magcius** [comment]
I don't believe DWM on Windows does anything with the SMPTE ST.2086 metadata during its compositing, which indeed means that at most, the metadata could be sent down the wire for the TV to do tonemapping with, but TVs do not have consistent handling of that metadata.

That said, you don't need to send any HDR metadata. For now, if you know your display's MaxTML, which you can query using DXGI, or by running an application like [VESA's DisplayHDR Test](https://apps.microsoft.com/store/detail/displayhdr-test/9NN1GPN70NF3?hl=en-us&gl=us&rtc=1), you can test your engine by doing your tonemapping relative to those levels.

**Comment by magcius** [comment]
The HDR metadata reported by the display is an optional EDID block (CTA-861-G section 7.5.13, HDR Static Metadata). The solution recommended in HGIG is to have a fallback database of TVs, and a calibration mechanism in case the data is missing.

I believe you should be able to use the [Windows HDR Calibration](https://support.microsoft.com/en-us/windows/calibrate-your-hdr-display-using-the-windows-hdr-calibration-app-f30f4809-3369-43e4-9b02-9eabebd23f19) app to configure the MinTML/MaxTML/MaxFFTML values reported by DXGI.

**Comment by versalinyaa** [comment]
Inside Khronos, we are actively discussing solutions to this problem with a renewed focus. We agree that this is an important problem that needs to be solved.

**Comment by MarijnS95** [comment]
For completeness, issue #1787 requests the same feature and can be closed as well when this is implemented and resolved.
